src/components/Toggle/Toggle.module.css

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+.toggle {
+  display: grid;
+  width: 44px;
+  height: 24px;
+}
+
+.toggle > * {
+  grid-area: 1/1;
+  width: 44px;
+  height: 24px;
+  box-sizing: border-box;
+}
+
+.toggle > input[type="checkbox"] {
+  opacity: 0;
+  margin: 0;
+  cursor: pointer;
+}
+
+.toggle > input[type="checkbox"][disabled] {
+  cursor: not-allowed;
+}
+
+.toggle-ui {
+  pointer-events: none;
+  width: 44px;
+  aspect-ratio: 1 / 1;
+  border-radius: 9999px;
+  border: 1px solid var(--cpd-color-border-interactive-primary);
+  position: relative;
+  padding: 1px;
+  transition: all 0.2s ease-in-out;
```

Comment:
From @janogarcia 

> Default to ease instead of ease-in-out for snappier interactions. Note: We haven't formalized the motion guidelines or any related tokens, but we could take this as an opportunity to formalize the first one.

```suggestion
  transition: all 0.2s ease;
```